### DISCUSSION TITLE
Knitro options

### DISCUSSION THREAD

**Discussion by richteropt** [casadi/casadi#3107] [General]
Hi!

I am missing the possibility to set _any_ Knitro option, e.g. blasoption or feastol_abs. A good and easy way would be via .opt files for Knitro. This feature would be very nice to have.

Thanks.

Best regards,
Stefan

**Comment by jgillis** [comment]
You should be able to pass the `knitro.feastol_abs` similar to https://github.com/casadi/casadi/wiki/FAQ%3A-how-to-pass-options-to-solvers

**Comment by jgillis** [comment]
Here are some knitro options passed in the test suite: https://github.com/casadi/casadi/blob/d927522ff85961db404474d6e107cf15a7ce6e7d/test/python/nlp.py#L69

**Comment by richteropt** [comment]
Hi Joris,
thanks for your quick answer. If I pass, e.g. the feastol_abs parameter via 

options.knitro.feastol_abs = 1e-6;
solver = nlpsol('solver', 'knitro', nlp, options);

I get the following: 

Error in Function::call for 'solver' [KnitroInterface] at .../casadi/core/function.cpp:1401:
Error in Function::call for 'solver' [KnitroInterface] at .../casadi/core/function.cpp:330:
.../casadi/interfaces/knitro/knitro_interface.cpp:241: Assertion "KN_get_param_id(m->kc, op.first.c_str(),
&param_id)==0" failed:
Unknown parameter 'feastol_abs'.

One good way would be to allow to pass in options via .opt files. Then this would also be independent of the used Knitro version.
BTW: Most of the Knitro parameters can be set without problems.

Many thanks.

Best regards,
Stefan

**Comment by jgillis** [comment]
Feel free to open an issue/pull request concerning option file reading. From CasADi point of view that's a nonstandard approach

**Comment by richteropt** [comment]
Sure. Thanks.

**Comment by jgillis** [comment]
Implemented in 3.6.1

**Reply by richteropt** [reply]
Great. Thanks!
